system/doc/design_principles/secure_coding.md

```diff
@@ -0,0 +1,1986 @@
+library. Such application parameters are only available after the application
+has been started using the `application:start/1` functionality. That is, in
+order to make sure that the `libcrypto` library is configured as expected, make
+sure not to load the `m:crypto` module by any other means than calling
+`application:start(crypto)`. Any calls to the `m:crypto` library will load the
+module. Make sure not to call any functions in the `m:crypto` module, including
+`crypto:start/0` prior to the call to `application:start(crypto)`.
+
+In order to ensure that the FIPS mode is configured correctly in the
+`libcrypto` library from OpenSSL, use the [`fips_mode`] application parameter
+instead of calling `crypto:enable_fips_mode/1`.
+
+[`fips_mode`]: `e:crypto:crypto_app.html#fips_mode`
+
+[](){: #cryptographically-secure-random-numbers }
+[Cryptographically Secure Random Numbers]: #cryptographically-secure-random-numbers
+#### Cryptographically Secure Random Numbers
+
+Do not use `crypto:rand_uniform/2` since it uses functionality from the
+OpenSSL `libcrypto` library that in old versions of the library does not
+produce cryptographically secure random numbers ([CWE-338]). An alternative is
+using `rand:uniform/1` together with a cryptographically secure random number
+generator like `crypto:rand_seed/0`. Note that `crypto:rand_seed/0` not
+only produce a seed but also selects a generator which will be used by
+`rand:uniform/1`.
+
+The above mentioned functionality will store a state in the
+[process dictionary] of currently executing process. This state could be
+modified to select another generator by other functionality called interleaved
+with calls retrieving random numbers. A better solution is to use the
+functional API which will pass the state between calls instead of storing it in
+the process dictionary. The functional API are functions in the `m:rand` and
+`m:crypto` modules with function names having the suffix `_s`. That is, you
+might want to consider using `crypto:rand_seed_s/0`/`rand:uniform_s/2`
+instead of `crypto:rand_seed/0`/`rand:uniform/1`.
+
+[process dictionary]: `e:system:ref_man_processes.html#process-dictionary`
+[CWE-338]: https://cwe.mitre.org/data/definitions/338.html
+
+#### Legacy `crypto` Functions
+
+RSA with PKCS-1 padding is weak and should be avoided: do not use the 
+`rsa_pkcs1_padding` option with `crypto:private_encrypt/4`,
+`crypto:private_decrypt/4`, `crypto:public_encrypt/4`, or
+`crypto:public_decrypt/4`.
+
+For signatures, use `crypto:sign/4`/`crypto:verify/5` instead.
+
+[](){: #ssl-application }
+[`ssl` Application]: #ssl-application
+[`ssl`]: `e:ssl:index.html`
+### [`ssl`] Application
+
+The [`ssl`] application is secure by default and remains so if the
+recommendations in its documentation is followed. One must be especially
+careful when enabling more functionality, as it weakens security. It should
+only be done when absolutely necessary.
+
```

Comment:
Maybe say enabling legacy functionality as also very new functionality might have to be enable when it is first added before it becomes part of the defaults.